web/src/components/forms/add-project-to-team-form.tsx

```diff
@@ -0,0 +1,89 @@
+import {Field, Form} from '@/components/form';
+import {useAuth} from '@/context/auth-provider';
+import {useIsAuthorisedTo} from '@/hooks/auth-hooks';
+import {modifyTeamForProject} from '@/hooks/project-hooks';
+import {useGetTeams} from '@/hooks/queries';
+import {Action} from '@faims3/data-model';
+import {useQueryClient} from '@tanstack/react-query';
+import {z} from 'zod';
+
+interface AddProjectToTeamFormProps {
+  setDialogOpen: React.Dispatch<React.SetStateAction<boolean>>;
+  projectId: string;
+}
+
+/**
+ * A form component that allows assigning a project to a team.
+ *
+ * @param setDialogOpen - A function to control the dialog's open state.
+ * @param projectId - The ID of the project to be assigned to a team.
+ */
+export function AddProjectToTeamForm({
+  setDialogOpen,
+  projectId,
+}: AddProjectToTeamFormProps) {
+  const {user} = useAuth();
+  const QueryClient = useQueryClient();
+  const teams = useGetTeams(user);
+
+  // can we add a user to the team?
+  const canAddProjectToTeam = useIsAuthorisedTo({
+    action: Action.CHANGE_PROJECT_TEAM,
+    resourceId: projectId,
+  });
+
+  const teamsAvailable = teams.data?.teams;
+
+  if (!canAddProjectToTeam || !teamsAvailable) {
+    return <></>;
+  }
```

Comment:
loading during teams fetch? these are different outcomes - one is an error (shouldn't be here!) the other is normal data fetching